src/urllib3/connectionpool.py

```diff
@@ -56,10 +56,10 @@
 
 log = logging.getLogger(__name__)
 
-_Default = object()
+_Default: float = cast(float, object())
 
 
-_TYPE_TIMEOUT = Union[Timeout, int, float, object]
+_TYPE_TIMEOUT = Union[Timeout, int, float]
```

Comment:
Unrelated to the PR: in python typing (unlike in runtime), `int` is actually a sub-type of `float`, so the `int` here is redundant. However it makes to keep for clarity.